src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -533,15 +543,25 @@ def in_bcache?(device)
       # The device can be a planned one or filesystem from the devicegraph.
       #
       # @param device [Filesystems::Base, Planned::Device, nil]
-      # @return [Boolean] false if device is nil
+      # @return [Boolean] device encryption state; return false if device is nil
       def encrypted?(device)
-        return false if device.nil?
+        !encryption_type(device).is?(:none)
+      end
 
+      # Encryption type of device
+      #
+      # The device can be a planned one or filesystem from the devicegraph.
+      #
+      # Note: returns EncryptionType::LUKS1 for Planned::Device as there's no encryption type.
+      #
+      # @param device [Filesystems::Base, Planned::Device, nil]
+      # @return [Y2Storage::EncryptionType] Encryption type
+      def encryption_type(device)
         if device.is_a?(Planned::Device)
-          device.respond_to?(:encrypt?) && device.encrypt?
-        else
-          device.plain_blk_devices.any? { |d| d.respond_to?(:encrypted?) && d.encrypted? }
-        end
+          (device.respond_to?(:encrypt?) && device.encrypt?) ? Y2Storage::EncryptionType::LUKS1 : nil
+        elsif device.respond_to?(:plain_blk_devices)
+          device.plain_blk_devices.map { |d| d.encryption&.type }.compact.first
+        end || Y2Storage::EncryptionType::NONE
```

Comment:
I though you wanted to kill this `if` by unifying the APIs of the descendants of `Y2Storage::Device` and the descendants of `Y2Storage::Planned::Device`.

Reply:
I would like to see it changed. But not as part of this rather specific pr.

Reply:
ok